pkg/database/database.go

```diff
@@ -137,6 +137,8 @@ type DB interface {
 
 	IsClosed() bool
 	Close() error
+
+	Truncate(rp time.Duration) error
```

Comment:
returning the transaction id up to which it was truncated may be of interest

Reply:
I'll address this in a different PR as this requires changes on the pkg/database level